### DISCUSSION TITLE
How to use the code of mpc

### DISCUSSION THREAD

**Discussion by 1191000814** [BUAA-BDA/OpenHuFu#184] [Q&A]
Hello, I want to use your code for the other problem of smpc, can you give me some advice to use it? Thanks a lot!

**Comment by SongY123** [comment]
If you encounter any issues or have specific questions about the project, please feel free to raise them in the discussion section or send me an email directly at [songyangbuaa@buaa.edu.cn](url). I will be happy to assist you.

**Comment by 1191000814** [comment]
您好, 我想要使用您的开源项目OpenHufu中的安全多方计算(mpc)部分去解决其他的问题, 我目前正在阅读您项目的源码, 关于如何使用您的SMPC方面的代码, 您有什么好的建议吗


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "BUAA-BDA/OpenHuFu"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年7月14日(星期天) 晚上11:20
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [BUAA-BDA/OpenHuFu] How to use the code of mpc (Discussion #184)





 
If you encounter any issues or have specific questions about the project, please feel free to raise them in the discussion section or send me an email directly at ***@***.*** I will be happy to assist you.

**Comment by SongY123** [comment]
我理解您想使用项目中的一些MPC算法，比如秘密共享去解决一些其他问题，项目里面的通信接口与表结构和拥有方是绑定的。您可以按照自己的需求去重写通信那块的代码，或者可以采用一些开源的MPC库，比如C/C++的MP-SPDZ、OblivC等，他们对基础的MPC算子支持更好。

**Comment by 1191000814** [comment]
我想要重构您的代码, 是不是只需要mpc, rpc这两个模块就行了吗, 想问下工作量会很大吗




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "BUAA-BDA/OpenHuFu"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年7月15日(星期一) 凌晨0:18
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [BUAA-BDA/OpenHuFu] How to use the code of mpc (Discussion #184)





 
我理解您想使用项目中的一些MPC算法，比如秘密共享去解决一些其他问题，项目里面的通信接口与表结构和拥有方是绑定的。您可以按照自己的需求去重写通信那块的代码，或者可以采用一些开源的MPC库，比如C/C++的MP-SPDZ、OblivC等，他们对基础的MPC算子支持更好。

**Reply by SongY123** [reply]
不太清楚您的需求，如果只是重写mpc的代码，一般来说涉及的就是mpc这个module，其他模块涉及的不会很多。
